### PR TITLE
ER-797 blocking accounts of non levy employers

### PR DESCRIPTION
**Pull request by sachpatel**
This is very crude job. I have not gone through the client interface as this is a temp job that shouldn't be baked into our client package.

You will need to request access to the following db's in order to run:

- das-test-eas-acc-db
- das-test-eas-fin-db

### CODE REVIEW

File: src/Jobs/Recruit.Vacancies.Jobs/NonLevyAccountBlocker/AccountsReader.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                {
+                    var command = new SqlCommand(SelectLevyPayerAccountsSql, conn);
+
+                    conn.Open();
+
+                    var reader = await command.ExecuteReaderAsync();
+
+                    while (await reader.ReadAsync())
+                    {
+                        //var x = reader[""].ToString();
```
Comment (chrisjensenuk) [inline src/Jobs/Recruit.Vacancies.Jobs/NonLevyAccountBlocker/AccountsReader.cs]: delete comment

File: src/Jobs/Recruit.Vacancies.Jobs/NonLevyAccountBlocker/AccountsReader.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            _accountsDbConnString = accountsDbConnString;
+        }
+
+        public async Task<IList<string>> GetLevyPayerAccountIds()
+        {
+            var accountIdentifiers = new List<string>();
+
+            try
+            {
+                using (var conn = new SqlConnection(_financeDbConnString))
```
Comment (chrisjensenuk) [inline src/Jobs/Recruit.Vacancies.Jobs/NonLevyAccountBlocker/AccountsReader.cs]: I know time is tight but...wrap this in a retry?

File: src/Jobs/Recruit.Vacancies.Jobs/NonLevyAccountBlocker/AccountsReader.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            return accountIdentifiers;
+        }
+
+        public async Task<IList<EmployerAccountIdentifier>> GetEmployerAccounts()
+        {
+            var hashedAccountIdentifiers = new List<EmployerAccountIdentifier>();
+
+            try
+            {
+                using (var conn = new SqlConnection(_accountsDbConnString))
```
Comment (chrisjensenuk) [inline src/Jobs/Recruit.Vacancies.Jobs/NonLevyAccountBlocker/AccountsReader.cs]: retry?

File: src/Jobs/Recruit.Vacancies.Jobs/NonLevyAccountBlocker/AccountsReader.cs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        private readonly string _accountsDbConnString;
+
+        public AccountsReader(ILogger<AccountsReader> logger, string financeDbConnString, string accountsDbConnString)
+        {
+            _logger = logger;
+            _financeDbConnString = financeDbConnString;
+            _accountsDbConnString = accountsDbConnString;
+        }
+
+        public async Task<IList<string>> GetLevyPayerAccountIds()
```
Comment (chrisjensenuk) [inline src/Jobs/Recruit.Vacancies.Jobs/NonLevyAccountBlocker/AccountsReader.cs]: GetLevyPayerAccountIds**Async**

File: src/Jobs/Recruit.Vacancies.Jobs/NonLevyAccountBlocker/AccountsReader.cs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error retrieving levy payer employer account id's from Finance DB.");
+                throw;
+            }
+
+            return accountIdentifiers;
+        }
+
+        public async Task<IList<EmployerAccountIdentifier>> GetEmployerAccounts()
```
Comment (chrisjensenuk) [inline src/Jobs/Recruit.Vacancies.Jobs/NonLevyAccountBlocker/AccountsReader.cs]: GetEmployerAccounts**Async**

**Comment by sachpatel** [comment]
@chrisjensenuk  the global filter was already suggested before. This will mean the previous testing is invalidated.

**Comment by mgwilliam** [comment]
@chrisjensenuk note this isn't the same as the employer blocking stuff you've done already. they are actually separate requirements which are temporarily being put together so i'd advise against changing anything you've already done. more detail in ER-798

**Pull request merged**